src/pathPlanning/Comb.cpp

```diff
@@ -106,7 +106,11 @@ bool Comb::calc(const ExtruderTrain& train, Point startPoint, Point endPoint, Co
     {
         PolygonsPart part = partsView_inside_optimal.assemblePart(start_part_idx);
         combPaths.emplace_back();
-        return LinePolygonsCrossings::comb(part, *inside_loc_to_line_optimal, startPoint, endPoint, combPaths.back(), -offset_dist_to_get_from_on_the_polygon_to_outside, max_comb_distance_ignored, fail_on_unavoidable_obstacles);
+        const bool combing_succeeded = LinePolygonsCrossings::comb(part, *inside_loc_to_line_optimal, startPoint, endPoint, combPaths.back(), -offset_dist_to_get_from_on_the_polygon_to_outside, max_comb_distance_ignored, fail_on_unavoidable_obstacles);
+        // If the endpoint of the travel path changes with combing, then it means that we are moving to an outer wall
+        // and we should unretract before the last travel move when travelling to that outer wall
+        unretract_before_last_travel_move = combing_succeeded && endPoint != travel_end_point_before_combing;
```

Comment:
This isn't very likely to occur, if it indeed _can_ occur, but is there a possibility of the end-point just _happening to be_ on that specific spot on an outer wall?

Reply:
Initially the end-point is indeed on the outer wall, but combing decides to move it on the inside to avoid blobs and inaccuracies on the outer wall. If combing decides to leave it on the outer wall, then it means that combing did not succeed to find a proper combing path. In such cases, it is reasonable to not unretract before the last travel move, as there is no combing either.